### ISSUE TITLE
no-missing-keys warns when using <i18n> blocks

### ISSUE THREAD

**Issue by 0gust1** [intlify/eslint-plugin-vue-i18n#70]
Thanks for the eslint plugin, it helps a lot !

On my project, I use a combination of global translation files and component-local `<i18n>` blocks. 
I still get lint warnings `eslint(vue-i18n/no-missing-keys)` on translation strings defined in  `<i18n>` blocks, and I don't understand why (everything work OK in the app).

Here our eslint config :
```
  "eslintConfig": {
    "root": true,
    "env": {
      "node": true
    },
    "extends": [
      "plugin:vue/essential",
      "@vue/standard",
      "plugin:vue-i18n/recommended",
      "plugin:chai-friendly/recommended"
    ],
    "rules": {
      "no-console": [
        "error",
        {
          "allow": [
            "warn",
            "error"
          ]
        }
      ]
    },
    "settings": {
      "vue-i18n": {
        "localeDir": "./src/locales/*.json"
      }
    },
    "parserOptions": {
      "parser": "babel-eslint"
    },
    "overrides": [
      {
        "files": [
          "**/__tests__/*.{j,t}s?(x)",
          "**/tests/unit/**/*.spec.{j,t}s?(x)"
        ],
        "env": {
          "jest": true
        }
      }
    ]
  }
```

Here, the dependencies (we used vue-cli) :
```
  "dependencies": {
    "core-js": "^3.6.5",
    "vue": "^2.6.10",
    "vue-i18n": "^8.18.2",
    "vue-router": "^3.3.4"
  },
  "devDependencies": {
    "@intlify/vue-i18n-loader": "^1.0.0",
    "@tailwindcss/ui": "^0.3.0",
    "@vue/cli-plugin-babel": "^4.4.6",
    "@vue/cli-plugin-e2e-cypress": "^4.4.6",
    "@vue/cli-plugin-eslint": "^4.4.6",
    "@vue/cli-plugin-router": "^4.4.6",
    "@vue/cli-plugin-unit-jest": "^4.4.6",
    "@vue/cli-service": "^4.4.6",
    "@vue/eslint-config-standard": "^5.1.2",
    "@vue/test-utils": "1.0.0-beta.29",
    "babel-eslint": "^10.1.0",
    "eslint": "^6.8.0",
    "eslint-plugin-chai-friendly": "^0.6.0",
    "eslint-plugin-import": "^2.22.0",
    "eslint-plugin-node": "^11.1.0",
    "eslint-plugin-promise": "^4.2.1",
    "eslint-plugin-standard": "^4.0.0",
    "eslint-plugin-vue": "^6.2.2",
    "eslint-plugin-vue-i18n": "^0.3.0",
    "vue-cli-plugin-i18n": "^1.0.1",
    "vue-cli-plugin-tailwind": "^1.4.1",
    "vue-template-compiler": "^2.6.10"
  }
```

**Comment by 0gust1** [comment]
One thing I noticed in the eslint output, is that every warning is doubled :
```
/PATH_TO_PROJ/frontend/src/components/ProjectsList.vue
  5:14  warning  'projectsList' does not exist  vue-i18n/no-missing-keys
  5:14  warning  'projectsList' does not exist  vue-i18n/no-missing-keys
```

I have the same phenomenon in VSCode "on-the-fly" linting.

**Comment by ota-meshi** [comment]
Hi @0gust1.
I don't know the details of your problem yet, but is PR #80 likely to solve it?

**Comment by mandragorn** [comment]
I'm using version v0.7.0 of eslint-plugin-vue-i18n (which includes PR #80 I believe) and I'm getting a similar error.

Simplified Test.vue:
```
<template>
  <i18n path="message_key" tag="p" />
</template>

<i18n>
{
  "en": {
    "message_key": "hi"
  }
}
</i18n>
```
.eslintrc.js:
```
module.exports = {
  root: true,
  env: {
    node: true
  },
  extends: [
    'plugin:vue/essential',
    'eslint:recommended',
    '@vue/prettier',
    'plugin:@intlify/vue-i18n/recommended'
  ],
  parserOptions: {
    parser: 'babel-eslint'
  },
  rules: {
    'no-console': process.env.NODE_ENV === 'production' ? 'warn' : 'off',
    'no-debugger': process.env.NODE_ENV === 'production' ? 'warn' : 'off',
    'prettier/prettier': [
      'warn',
      {
        singleQuote: true,
        trailingCommas: 'none',
        vueIndentScriptAndStyle: true
      }
    ],
    '@intlify/vue-i18n/no-html-messages': 'error',
    '@intlify/vue-i18n/no-missing-keys': 'error',
    '@intlify/vue-i18n/no-raw-text': 'error',
    '@intlify/vue-i18n/no-v-html': 'error',
    '@intlify/vue-i18n/no-dynamic-keys': 'error',
    '@intlify/vue-i18n/no-unused-keys': 'error'
  },
  overrides: [
    {
      files: [
        '**/__tests__/*.{j,t}s?(x)',
        '**/tests/unit/**/*.spec.{j,t}s?(x)'
      ],
      env: {
        jest: true
      }
    }
  ],
  settings: {
    'vue-i18n': {
      localeDir: {
        pattern: './*.vue',
        localeKey: 'key'
      }
    }
  }
};

```

Error when running lint:
```
error: unused 'en.message_key' key (@intlify/vue-i18n/no-unused-keys) at src/components/Test.vue:8:5:
   6 | {
   7 |   "en": {
>  8 |     "message_key": "hi"
     |     ^
   9 |   }
  10 | }
  11 | </i18n>

```

Dependencies:
```

  "dependencies": {
    "axios": "^0.18.0",
    "core-js": "^3.6.5",
    "moment": "^2.27.0",
    "v-mask": "^2.2.3",
    "vue": "^2.6.11",
    "vue-i18n": "^8.20.0",
    "vue-router": "^3.2.0",
    "vuejs-logger": "^1.5.3",
    "vuelidate": "^0.7.5",
    "vuex": "^3.4.0"
  },
  "devDependencies": {
    "@intlify/eslint-plugin-vue-i18n": "^0.7.0",
    "@intlify/vue-i18n-loader": "^1.0.0",
    "@kazupon/vue-i18n-loader": "^0.5.0",
    "@vue/cli-plugin-babel": "~4.4.0",
    "@vue/cli-plugin-e2e-cypress": "~4.4.0",
    "@vue/cli-plugin-eslint": "~4.4.0",
    "@vue/cli-plugin-pwa": "^4.4.6",
    "@vue/cli-plugin-router": "~4.4.0",
    "@vue/cli-plugin-unit-jest": "~4.4.0",
    "@vue/cli-plugin-vuex": "~4.4.0",
    "@vue/cli-service": "~4.4.0",
    "@vue/eslint-config-prettier": "^6.0.0",
    "@vue/test-utils": "^1.0.3",
    "babel-eslint": "^10.1.0",
    "eslint": "^6.8.0",
    "eslint-plugin-prettier": "^3.1.3",
    "eslint-plugin-vue": "^6.2.2",
    "lint-staged": "^9.5.0",
    "node-sass": "^4.12.0",
    "npm-force-resolutions": "0.0.3",
    "prettier": "^1.19.1",
    "register-service-worker": "^1.7.1",
    "sass-loader": "^8.0.2",
    "vue-cli-plugin-i18n": "~1.0.1",
    "vue-template-compiler": "^2.6.11",
    "workbox-webpack-plugin": "^5.1.3"
  },
```

**Comment by mandragorn** [comment]
I just realized this is the wrong issue for this, I will open up another issue.  Here I'm using the <i18n> vue _component_ and the path is not getting validated against the <i18n> block in a single file component.

**Comment by ota-meshi** [comment]
@0gust1 Try using a new version that supports `<i18n>` component.

**Comment by 0gust1** [comment]
@ota-meshi thanks a lot for the info and your patience ! 🙏  👍

**Comment by 0gust1** [comment]
Project dependencies updated, the problem keeps happening, sorry :-/

The error message has changed after deps updates :
```
   warning: 'password' does not exist in 'en' (@intlify/vue-i18n/no-missing-keys) at src/views/Home.vue:58:21:
  56 |                   for="password"
  57 |                   class="block text-sm font-medium leading-5 text-gray-700"
> 58 |                 >{{ $t("password") }}</label>
     |                     ^
  59 |                 <div class="mt-1 rounded-md shadow-sm">
  60 |                   <input
  61 |                     v-model="password"
```

relevant part of `package.json`
```json
{
  "devDependencies": {
    "@intlify/eslint-plugin-vue-i18n": "^0.8.1",
    "@intlify/vue-i18n-loader": "^1.0.0",
    "@tailwindcss/ui": "^0.3.0",
    "@vue/cli-plugin-babel": "^4.5.3",
    "@vue/cli-plugin-e2e-cypress": "^4.5.3",
    "@vue/cli-plugin-eslint": "^4.5.3",
    "@vue/cli-plugin-router": "^4.5.3",
    "@vue/cli-plugin-unit-jest": "^4.5.3",
    "@vue/cli-service": "^4.5.3",
    "@vue/eslint-config-standard": "^5.1.2",
    "@vue/test-utils": "1.0.0-beta.29",
    "babel-eslint": "^10.1.0",
    "eslint": "^6.8.0",
    "eslint-plugin-chai-friendly": "^0.6.0",
    "eslint-plugin-import": "^2.22.0",
    "eslint-plugin-node": "^11.1.0",
    "eslint-plugin-promise": "^4.2.1",
    "eslint-plugin-standard": "^4.0.0",
    "eslint-plugin-vue": "^6.2.2",
    "vue-cli-plugin-i18n": "^1.0.1",
    "vue-cli-plugin-tailwind": "^1.4.1",
    "vue-template-compiler": "^2.6.10"
  },
  "eslintConfig": {
    "root": true,
    "env": {
      "node": true
    },
    "extends": [
      "plugin:vue/essential",
      "@vue/standard",
      "plugin:@intlify/vue-i18n/recommended",
      "plugin:chai-friendly/recommended"
    ],
    "rules": {
      "no-console": [
        "error",
        {
          "allow": [
            "warn",
            "error"
          ]
        }
      ]
    },
    "settings": {
      "vue-i18n": {
        "localeDir": "./src/locales/*.json"
      }
    },
    "parserOptions": {
      "parser": "babel-eslint",
      "sourceType": "module"
    },
    "overrides": [
      {
        "files": [
          "**/__tests__/*.{j,t}s?(x)",
          "**/tests/unit/**/*.spec.{j,t}s?(x)"
        ],
        "env": {
          "jest": true
        }
      }
    ]
  }
}
```

`vue.config.js` :
```js
module.exports = {
  // outputDir: path.resolve(__dirname, '../backend/public'),

  devServer: {
    proxy: 'http://localhost:3000'
  },

  pluginOptions: {
    i18n: {
      locale: 'en',
      fallbackLocale: 'fr',
      localeDir: 'locales',
      enableInSFC: true
    }
  }
}
```
